### PR TITLE
Adds more TCs for Tree test

### PR DESCRIPTION
**Pull request by humdrum**
**What this PR does / why we need it**:
- Adds more TCs for Tree Tests.
  -  Convert all TCs for Tree in JS to iOS
- Fix Bugs
  -  Change the misbehaved delete() in ElementRHT
  - Add missing information when cloning CRDTTree instance.
  - print with quotations with primitive values when making XML  output.
- Change function names of CRDTContainer
  - delete -> purge
  - remove(createdAt) -> delete
  - remove(index) ->  deleteByIndex
 
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

File: Sources/Document/Json/JSONArray.swift [hunk 1]
```diff
@@ -381,7 +381,7 @@ public class JSONArray {
     @discardableResult
     private func removeInternal(byIndex index: Int) -> CRDTElement? {
         let ticket = self.context.issueTimeTicket
-        let removed = try? self.target.remove(index: index, executedAt: ticket)
+        let removed = try? self.target.deleteByIndex(index: index, executedAt: ticket)
```
Comment (myupage) [inline Sources/Document/Json/JSONArray.swift:384]: Using `guard` will keep the code shorter

```swift        
guard let removed = try? self.target.deleteByIndex(index: index, executedAt: ticket) else {
     return nil
}
```

File: Sources/Document/Json/JSONObject.swift [hunk 2]
```diff
@@ -206,10 +206,13 @@ public class JSONObject {
...
     }
 
+    /**
+     * `remove` deletes the value of the given key.
+     */
     public func remove(key: String) {
         Logger.trace("obj[\(key)]")
 
-        let removed = try? self.target.remove(key: key, executedAt: self.context.issueTimeTicket)
+        let removed = try? self.target.deleteByKey(key: key, executedAt: self.context.issueTimeTicket)
```
Comment (myupage) [inline Sources/Document/Json/JSONObject.swift:215]: Same here. Consider adding `guard`.

File: Sources/Document/Json/ObjectDataHandler.swift [hunk 3]
```diff
@@ -165,7 +165,7 @@ class ObjectDataHandler {
     }
 
     func remove(key: String) throws {
-        let removed = try? self.target.remove(key: key, executedAt: self.context.issueTimeTicket)
+        let removed = try? self.target.deleteByKey(key: key, executedAt: self.context.issueTimeTicket)
```
Comment (myupage) [inline Sources/Document/Json/ObjectDataHandler.swift:168]: And here.

File: Sources/Document/Json/JSONArray.swift (hunk 1 above)
Comment (humdrum) [reply Sources/Document/Json/JSONArray.swift]: Thanks.

**Pull request merged**